src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -270,7 +287,7 @@ public IAsyncCursor<TResult> Execute(IReadBinding binding, CancellationToken can
             using (var channelBinding = new ChannelReadBinding(channelSource.Server, channel, binding.ReadPreference, binding.Session.Fork()))
             {
                 cursor = Resume(channelBinding, cancellationToken);
-                if (_startAtOperationTime == null && _resumeAfter == null)
+                if (_startAtOperationTime == null && _resumeAfter == null && _startAfter == null)
```

Comment:
Add a new helper method ExecuteAggregateOperation that takes a resuming parameter indicating whether we are executing the first aggregate command or one to resume the change stream. Also, refactor lines 290-297 to a helper method.

So lines 289-297 become:

    cursor = ExecuteAggregateOperation(channelBinding, resuming: false, cancellationToken);
    SaveInitialOperationTimeIfRequired(channel, channelBinding, cursor);

Reply:
The new helper method would be:

    private void SaveInitialOperationTimeIfRequired(IChannelHandle channel, IReadBinding binding, IAsyncCursor<RawBsonDocument> cursor)
    {
        if (_startAtOperationTime == null && _resumeAfter == null && _startAfter == null)
        {
            var maxWireVersion = channel.ConnectionDescription.IsMasterResult.MaxWireVersion;
            if (maxWireVersion >= 7)
            {
                if (_documentResumeToken != null)
                {
                    _initialOperationTime = binding.Session.OperationTime;
                }
            }
        }
    }

Note the *new* check that _documentResumeToken != null. That's from the spec.

Reply:
Done